website/docs/r/data_protection_backup_instance_postgresql.html.markdown

```diff
@@ -0,0 +1,110 @@
+
+-> **Note**: Before using this resource, there are some prerequisite permissions for configure backup and restore. See more details from https://docs.microsoft.com/en-us/azure/backup/backup-azure-database-postgresql#prerequisite-permissions-for-configure-backup-and-restore.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "rg" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_postgresql_server" "example" {
+  name                = "example-postgresql-server"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
+  sku_name = "B_Gen5_2"
+
+  storage_mb                   = 5120
+  backup_retention_days        = 7
+  geo_redundant_backup_enabled = false
+  auto_grow_enabled            = true
+
+  administrator_login          = "psqladminun"
+  administrator_login_password = "H@Sh1CoR3!"
+  version                      = "9.5"
+  ssl_enforcement_enabled      = true
+}
+
+resource "azurerm_postgresql_database" "example" {
+  name                = "example-postgresql-database"
+  resource_group_name = azurerm_resource_group.example.name
+  server_name         = azurerm_postgresql_server.example.name
+  charset             = "UTF8"
+  collation           = "English_United States.1252"
+}
+
+resource "azurerm_data_protection_backup_vault" "example" {
+  name                = "example-backup-vault"
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = azurerm_resource_group.rg.location
+  datastore_type      = "VaultStore"
+  redundancy          = "LocallyRedundant"
+}
+
+resource "azurerm_data_protection_backup_policy_postgresql" "example" {
+  name                            = "example-backup-policy"
+  resource_group_name             = azurerm_resource_group.rg.name
+  vault_name                      = azurerm_data_protection_backup_vault.example.name
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
+  default_retention_duration      = "P4M"
+}
+
+resource "azurerm_data_protection_backup_instance_postgresql" "example" {
+  name     = "example-backup-instance"
+  location = azurerm_resource_group.rg.location
+  vault_id = azurerm_data_protection_backup_vault.example.id
+
+  database_id      = azurerm_postgresql_database.example.id
+  backup_policy_id = azurerm_data_protection_backup_policy_postgresql.example.id
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Backup Instance PostgreSQL. Changing this forces a new Backup Instance PostgreSQL to be created.
+
+* `location` - (Required) The location of the source database. Changing this forces a new Backup Instance PostgreSQL to be created.
+
+* `vault_id` - (Required) The id of the Backup Vault where the Backup Instance PostgreSQL should exist.. Changing this forces a new Backup Instance PostgreSQL to be created.
```

Comment:
_minor_ but this can become:

```suggestion
* `vault_id` - (Required) The ID of the Backup Vault within which the PostgreSQL Backup Instance should exist. Changing this forces a new Backup Instance PostgreSQL to be created.
```